src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationScopeParsingTests.cs

```diff
@@ -1546,23 +1847,192 @@ public void Local_05(LanguageVersion langVersion)
                         N(SyntaxKind.SemicolonToken);
                     }
                 }
+                N(SyntaxKind.IncompleteMember);
+                {
+                    N(SyntaxKind.RefType);
+                    {
+                        N(SyntaxKind.RefKeyword);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "b");
+                        }
+                    }
+                }
+                N(SyntaxKind.GlobalStatement);
+                {
+                    N(SyntaxKind.EmptyStatement);
+                    {
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void TypeNamedScoped(LanguageVersion langVersion)
+        {
+            string source = @"
+class scoped { }
+";
+            // Missing error
+            // Tracked by https://github.com/dotnet/roslyn/issues/62931
+            UsingTree(source, TestOptions.Regular.WithLanguageVersion(langVersion));
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.ClassDeclaration);
+                {
+                    N(SyntaxKind.ClassKeyword);
+                    N(SyntaxKind.IdentifierToken, "scoped");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void TypeNamedScoped_Escaped(LanguageVersion langVersion)
+        {
+            string source = @"
+class @scoped { }
+";
+            UsingTree(source, TestOptions.Regular.WithLanguageVersion(langVersion));
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.ClassDeclaration);
+                {
+                    N(SyntaxKind.ClassKeyword);
+                    N(SyntaxKind.IdentifierToken, "@scoped");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void Local_06(LanguageVersion langVersion)
+        {
+            string source =
+@"scoped.nested a;
```

Comment:
>scoped.nested [](http://example.com/codeflow?start=2&length=13)

It seems odd that `scoped.nested` is treated as a type name in `scoped.nested a;` but not in `ref scoped.nested b;`.

Reply:
Filed follow-up issue: [https://github.com/dotnet/roslyn/issues/62950](https://github.com/dotnet/roslyn/issues/62950)